### ISSUE TITLE
Missing Transform_Fn implementation

### ISSUE THREAD

**Issue by mango-picket** [aws/sagemaker-scikit-learn-container#172]
Hi,

I am trying to deploy an SKLearnModel with custom inference script. In my script, if I define the input_fn, predict_fn, and output_fn, then everything works fine, but when I replace these three functions with the transform_fn, the inference defaults to the default input/predict/output_fn.

Upon inspection, I found the issue in the src/sagemaker_sklearn_container/serving.py 

```
def _user_module_transformer(user_module):
    model_fn = getattr(user_module, 'model_fn', default_model_fn)
    input_fn = getattr(user_module, 'input_fn', default_input_fn)
    predict_fn = getattr(user_module, 'predict_fn', default_predict_fn)
    output_fn = getattr(user_module, 'output_fn', default_output_fn)

    return transformer.Transformer(model_fn=model_fn, input_fn=input_fn, predict_fn=predict_fn,
                                   output_fn=output_fn)
```

This function doesnt wrap user module's transform_fn.
For example, the XGB_container  allows for the following:

```
def _user_module_transformer(user_module):
    model_fn = getattr(user_module, "model_fn", default_model_fn)
    input_fn = getattr(user_module, "input_fn", None)
    predict_fn = getattr(user_module, "predict_fn", None)
    output_fn = getattr(user_module, "output_fn", None)
    transform_fn = getattr(user_module, "transform_fn", None)

    if transform_fn and (input_fn or predict_fn or output_fn):
        raise exc.UserError("Cannot use transform_fn implementation with input_fn, predict_fn, and/or output_fn")

    if transform_fn is not None:
        return transformer.Transformer(model_fn=model_fn, transform_fn=transform_fn)
    else:
        return transformer.Transformer(
            model_fn=model_fn,
            input_fn=input_fn or default_input_fn,
            predict_fn=predict_fn or default_predict_fn,
            output_fn=output_fn or default_output_fn,
        )
```
Is there a reason to not allow transform_fn in scikit-learn containers?
If not, Can we please add this functionality?
I believe the above changes should suffice, but would love to hear back on this.

**Comment by mabunday** [comment]
Hi @mango-picket I just merged the change in. Can you let me know what region you're in? That way I can give you an estimate for when the deployment will complete and be available for your use.

**Comment by mango-picket** [comment]
I am trying to deploy the endpoints in US-West-2 and US-East-2 regions

**Comment by mabunday** [comment]
Hi, the change should be available in those regions now. Can you please confirm when you get a chance? Thank you!

**Comment by mango-picket** [comment]
> Hi, the change should be available in those regions now with the `1.2-1` container. Can you please confirm when you get a chance? Thank you!

Hi, Thanks for your quick responses, The transform_fn now works with 1.2-1 container for SkLearn.
This will help me handle errors more gracefully.
